# num 9

##### From Rodya 1


Откройте файл электронной таблицы, содержащей в каждой строке семь натуральных чисел. Определите количество строк таблицы, для чисел которых выполнены оба условия:
- в строке есть два числа, каждое из которых повторяется дважды, остальные три числа различны;
- среднее арифметическое всех повторяющихся чисел строки меньше среднего арифметического всех её чисел.

В ответе запишите только число.

In [ ]:
strings = [[int(x) for x in i.split()] for i in open("tasks/9rodya1.txt").readlines()]
cnt = 0

for string in strings:
    povt = [i for i in string if string.count(i) == 2]
    nepovt = [i for i in string if string.count(i) == 1]
    if len(povt) == 4:
        arpovt = sum(povt) / 4
        arnepovt = sum(nepovt) / 3
        if arpovt < arnepovt:
            cnt += 1

print(cnt)

Ответ: 83

##### № 27764 Апробация 04.03.26 

Откройте файл электронной таблицы, содержащий в каждой строке пять натуральных чисел. Определите количество строк таблицы, для которых выполнены оба условия:
- в строке все числа различны;
- удвоенная сумма максимального и минимального чисел строки равна сумме оставшихся трёх её чисел.

В ответе запишите только число.

In [ ]:
lines = [[int(x) for x in line.split()] for line in open("tasks/9a.txt").readlines()]
cnt = 0
for s in lines:
    max_s = max(s)
    min_s = min(s)
    s_ost = [x for x in s if x != max_s and x != min_s]
    if len(set(s)) == len(s) and 2 * (max_s + min_s) == sum(s_ost):
        cnt += 1
print(cnt)

Ответ: 5019

##### № 27621 Апробация 04.03.26

Откройте файл электронной таблицы, содержащей в каждой строке пять натуральных чисел. Определите наименьший номер строки таблицы, для которой выполнены оба условия:
- в строке все числа различны;
- разность максимального и минимального чисел строки равна сумме трёх её оставшихся чисел.

В ответе запишите только число.

In [ ]:
lines = [[int(x) for x in line.split()] for line in open("tasks/9b.txt").readlines()]
cnt = 0
for s in lines:
    cnt += 1
    max_s = max(s)
    min_s = min(s)
    s_ost = [x for x in s if x not in [max_s, min_s]]
    if len(set(s)) == len(s) and max_s - min_s == sum(s_ost):
        print(cnt)
        break

Ответ: 1321

##### № 27287 (Уровень: Базовый)

В файле электронной таблицы в каждой строке записаны семь целых чисел. Определите наибольшее повторяющееся число в строке таблице с наименьшим номером, для которой выполнены оба условия:
- в строке два разных числа повторяются трижды, одно число не повторяется;
- неповторяющееся число не больше минимального из повторяющихся чисел строки.

В ответе запишите абсолютное значение полученного результата.

In [ ]:
lines = [[int(x) for x in line.split()] for line in open("tasks/9_27287.txt")]
num = 0
for s in lines:
    povt = set([x for x in s if s.count(x) == 3])
    nepovt = [x for x in s if x not in povt]
    if len(povt) == 2 and max(nepovt) <= min(povt):
        print(max(povt))
        break

Ответ: 43

##### № 29962 Апробация 14.05.26 (Уровень: Базовый)

Откройте файл электронной таблицы, содержащей в каждой строке семь натуральных чисел. Определите наибольший номер строки таблицы, для чисел которой выполнены оба условия:
- в строке одно число повторяется трижды, остальные четыре числа различны;
- среднее арифметическое неповторяющихся чисел строки больше её повторяющего числа.

В ответе запишите только число.

In [4]:
lines = [[int(x) for x in line.split()] for line in open("tasks/num9_1.txt").readlines()]
cnt = 0

for line in lines:
    povt = [x for x in line if line.count(x) == 3]
    nepovt = [x for x in line if line.count(x) == 1]
    cnt += 1
    if len(nepovt) == 4:
        povtd = povt[0]
        sr_ar_nepovt = sum(nepovt) / 4
        if sr_ar_nepovt > povtd:
            num = cnt
print(num)

13609


Ответ: 13609

##### № 29341 Открытый вариант 2026 (Уровень: Базовый)

Откройте файл электронной таблицы, содержащей в каждой строке четыре натуральных числа. Определите количество строк таблицы, содержащих числа, для которых выполнены оба условия:
- наибольшее из четырёх чисел меньше суммы трёх других;
- четыре числа нельзя разбить на две пары чисел с равными суммами.

В ответе запишите только число.

In [7]:
lines = [[int(x) for x in line.split()] for line in open("tasks/num9_2.txt").readlines()]
cnt = 0

for line in lines:
    if len(set(line)) != 1:
        line = sorted(line)
        if (line[-1] < sum(line[:3])) and (sum(line[1:3]) != (line[0] + line[-1])):
            cnt += 1

print(cnt)

2354


Ответ: 2354

##### № 28930 ЕГКР 18.04.26 (Уровень: Базовый)

Откройте файл электронной таблицы, содержащей в каждой строке пять целых чисел. Определите количество строк таблицы, для которых выполнены оба условия:
- в строке все числа расположены в порядке возрастания;
- сумма минимального и максимального чисел строки не больше суммы оставшихся трёх её чисел.

В ответе запишите только число.

In [ ]:
lines = [[int(x) for x in line.split()] for line in open("tasks/num9_3.txt")]
cnt = 0

for line in lines:
    if line == sorted(line) and len(line) == len(set(line)):
        if line[0] + line[-1] <= sum(line[1:-1]):
            cnt += 1
print(cnt)

138


Ответ: 138

##### № 28755 Досрочная волна 2026 (Уровень: Базовый)

Откройте файл электронной таблицы, содержащей в каждой строке четыре натуральных числа. Определите количество строк таблицы, для чисел которых выполнены оба условия:
- наибольшее число строки меньше суммы трёх других;
- четыре числа нельзя разбирать на две пары с равными суммами.

В ответе запишите только число.

In [2]:
lines = [[int(x) for x in line.split()] for line in open("tasks/num9_4.txt")]
cnt = 0

for line in lines:
    line = sorted(line)
    if line[-1] < sum(line[:-1]) and (line[-1] + line[0]) != sum(line[1:-1]):
        cnt += 1

print(cnt)

2500


Ответ: 2500